### DISCUSSION TITLE
Embedded Documentation Link 404 Not Found

### DISCUSSION THREAD

**Discussion by darrenwoods3** [netbox-community/netbox#9709] [Q&A]
I have added DOCS_ROOT = '/netbox/docs' to the configuration.py file and still get the 404 Not Found message.  Any help would be greatly appreciated.

404 Not Found
nginx/1.18.0 (Ubuntu)

**Comment by candlerb** [comment]
* What version of Netbox are you running?
* Have you changed anything else significant in configuration.py - in particular have you changed `BASE_PATH`?  If so, have you made corresponding changes to your nginx config?
* What page are you accessing when you get that 404 error (what's visible in the URL bar?)
* What do you mean by "still get"?  Was it not working before you started making config changes?

If you install Netbox in the standard way, the documentation will be available in the standard location without changing any configs.  In my case, if I click on the "Docs" link on the home page, it takes me to `https://netbox.example.net/static/docs/` and I see the documentation as expected.

Note that [`DOCS_ROOT`](https://netbox.home.deploy2.net/static/docs/configuration/optional-settings/#docs_root) is a filesystem path, not a URL path.  Putting `/netbox/docs` implies that you've installed netbox in the root directory of your filesystem.  If you're not using the default location of `/opt/netbox` then it can work: you'll have other things to change (e.g. paths in your systemd unit file for a start).  Even then, you don't need to set `DOCS_ROOT` because the default value is relative to where you installed netbox.

**Comment by darrenwoods3** [comment]
Brian,

Thanks for the reply.


  *   Running version 3.2.5
  *   Only changes I have made to the config are the required and EMAIL, LOGIN_REQUIRED, RELEASE_CHECK_URL, TIME_ZONE and added the following to get fqdn to work.
CSRF_TRUSTED_ORIGINS = (
    'http://netbox.alamo.edu',
    'https://netbox.alamo.edu',
)

  *   This is a brand new install, BASE_PATH has not been changed BASE_PATH = ''
  *   Visible URL is http://netbox.alamo.edu/static/docs/
  *   It has not worked at all


***@***.***<https://www.alamo.edu/>

Darren L. Woods
Lead Enterprise Network Analyst
Information Technology Services
Alamo Colleges District
2222 N. Alamo | San Antonio, TX 78215
210-485-0508 (Direct) | 210-485-0000 (Main) | alamo.edu<https://www.alamo.edu/>

***@***.***


***@***.***

**Reply by candlerb** [reply]
> It has not worked at all

OK, then something has gone wrong.  First, remove `DOCS_ROOT` from your configuration.py to make things as close as possible to defaults.

Next: do you have any files in `/opt/netbox/netbox/static/docs/`?  If not, then you probably missed the step of [running the upgrade script](https://docs.netbox.dev/en/stable/installation/3-netbox/#run-the-upgrade-script), or there was an error which occurred while this ran which you did not notice.  Run it again, and check the output.

If you *do* have those files, then it probably means your nginx config is wrong. There is a working one [here](https://github.com/netbox-community/netbox/blob/develop/contrib/nginx.conf).

**Comment by darrenwoods3** [comment]
Brian there was not docs directory in /opt/netbox/netbox/static.  I ran the upgrade.sh script and that resolved this issue.  Thank you so much for troubleshooting and helping me get this resolved.



***@***.***<https://www.alamo.edu/>

Darren L. Woods
Lead Enterprise Network Analyst
Information Technology Services
Alamo Colleges District
2222 N. Alamo | San Antonio, TX 78215
210-485-0508 (Direct) | 210-485-0000 (Main) | alamo.edu<https://www.alamo.edu/>

***@***.***


***@***.***
